main.cpp

```diff
@@ -12,25 +12,34 @@ int main() {
 
     // Show the network address
     const char *ip = net.get_ip_address();
-    printf("IP address is: %s\n", ip ? ip : "No IP");
+    const char *netmask = net.get_netmask();
+    const char *gateway = net.get_gateway();
+    printf("IP address: %s\n", ip ? ip : "None");
+    printf("Netmask: %s\n", netmask ? netmask : "None");
+    printf("Gateway: %s\n", gateway ? gateway : "None");
 
     // Open a socket on the network interface, and create a TCP connection to mbed.org
     TCPSocket socket;
     socket.open(&net);
-    socket.connect("developer.mbed.org", 80);
+    socket.connect("api.ipify.org", 80);
+    char *buffer = new char[256];
```

Comment:
> but we probably shouldn't be encouraging stack allocations for buffers. 

Forgive my ignorance here. Is it not common practice to put buffers on stacks? Or I guess since it has the chance of buffer overruns it's not so bueno?